pkg/core/terraform/tf.go

```diff
@@ -155,33 +157,70 @@ func (sw *StdWriter) GetString() string {
 	return s
 }
 
-func (tf Terraform) formatTerraformWorkspaces(list string) (string) {
+func (tf Terraform) formatTerraformWorkspaces(list string) string {
 
 	list = strings.TrimSpace(list)
 	char_replace := strings.NewReplacer("*", "", "\n", ",", " ", "")
 	list = char_replace.Replace(list)
 	return list
 }
 
+func copyGitHubVars() (map[string]string, error) {
+	githubVars := os.Getenv("GITHUB_VARS")
+	var result = make(map[string]string)
+	if githubVars != "" {
+		parsedJson := make(map[string]string)
+		err := json.Unmarshal([]byte(githubVars), &parsedJson)
+		if err != nil {
+			return nil, err
+		}
+
+		for k, v := range parsedJson {
+			err := os.Setenv(k, v)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return result, nil
+}
+
 func (tf Terraform) Plan(params []string, envs map[string]string) (bool, string, string, error) {
-	workspaces, _, _, err := tf.runTerraformCommand("workspace", envs, "list")
+	githubVars, _ := copyGitHubVars()
```

Comment:
same as before, also why do we even need this?

Reply:
similar to secrets we need to get json with all github vars and set env vars accordingly, another option would be to map every var individually in github workflow's config

Reply:
why do we need all vars?

Reply:
for promotion they will just need to set one variable right? ENV_NAME or something. Setting all variables is an overkill

Reply:
that would allow to map github vars to terraform vars individually without tfvars file